### PR TITLE
support for esp32

### PR DESCRIPTION
**Pull request by donRaphaco**
relates to https://github.com/micro-ROS/micro-ros-build/pull/140

### CODE REVIEW

File: apps/ping_pong/app-colcon.meta [hunk 1]
```diff
@@ -9,8 +9,7 @@
                 "-DRMW_UXRCE_MAX_SERVICES=0",
                 "-DRMW_UXRCE_MAX_CLIENTS=0",
                 "-DRMW_UXRCE_MAX_HISTORY=4",
-                "-DRMW_UXRCE_DEFAULT_SERIAL_DEVICE=3"
```
Comment (donRaphaco) [inline apps/ping_pong/app-colcon.meta:12]: This option gets already set when the [configure.sh](https://github.com/micro-ROS/micro-ros-build/blob/26e9b778aa5067da3578210b3ddac2bfe0086147/micro_ros_setup/config/freertos/olimex-stm32-e407/configure.sh#L36) script is run with `-t serial`.
If set here it overrides the setting of the configure script.

File: apps/ping_pong/app.c [hunk 2]
```diff
@@ -1,5 +1,3 @@
-#include <allocators.h>
```
Comment (donRaphaco) [inline apps/ping_pong/app.c:1]: unused include

File: apps/int32_publisher/app.c [hunk 3]
```diff
@@ -7,6 +8,11 @@
 #include <rclc/rclc.h>
 #include <rclc/executor.h>
 
+#ifdef ESP_PLATFORM
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#endif
```
Comment (donRaphaco) [inline apps/int32_publisher/app.c:14]: for esp32 these includes are needed for `vTaskDelete`

File: apps/int32_publisher/app.c [hunk 4]
```diff
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <unistd.h>
```
Comment (donRaphaco) [inline apps/int32_publisher/app.c:2]: needed for `usleep` function

File: microros_esp32_extensions/sdkconfig.defaults [hunk 5]
```diff
@@ -0,0 +1,3 @@
+CONFIG_ESP_MAIN_TASK_STACK_SIZE=25000
+CONFIG_FREERTOS_UNICORE=y
+CONFIG_ESP_TASK_WDT=n
```
Comment (pablogs9) [inline microros_esp32_extensions/sdkconfig.defaults:3]: Hello @donRaphaco, do you have any idea on why is this necessary?

Comment (donRaphaco) [reply microros_esp32_extensions/sdkconfig.defaults:3]: The `CONFIG_ESP_TASK_WDT=n` setting I added to prevent getting a warning (during runtime) that the watchdog timer for the main task did not reset. I figured task watchdog timers aren't really useful if there is only one task, hence I deactivated them with this setting. ([about the esp32 task watchdog timer](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/wdts.html#task-watchdog-timer))

The setting `CONFIG_FREERTOS_UNICORE=y` I don't know why it's necessary for microros, but if I don't use this setting I'm getting a runtime error and I can't really figure out why exactly.

Comment (pablogs9) [reply microros_esp32_extensions/sdkconfig.defaults:3]: Thank you very much @donRaphaco

Comment (donRaphaco) [reply microros_esp32_extensions/sdkconfig.defaults:3]: @pablogs9 I just realized, that the error I had with `CONFIG_FREERTOS_UNICORE=y` is now solved somehow, which is really cool since I noticed a performance boost right away when I removed the setting.
So `CONFIG_FREERTOS_UNICORE=y` isn't actually necessary any more and should be removed :)

Comment (pablogs9) [reply microros_esp32_extensions/sdkconfig.defaults:3]: Thank you very much @donRaphaco. Could you share a reference link and I will commit the changes?

BTW, have you seen this: https://github.com/micro-ROS/micro_ros_espidf_component ?

Comment (donRaphaco) [reply microros_esp32_extensions/sdkconfig.defaults:3]: @pablogs9 what do you mean by "reference link"?

Yeah I've seen https://github.com/micro-ROS/micro_ros_espidf_component and tested it. I really like it, since it's better integrates with the esp-idf stuff 👍

Comment (pablogs9) [reply microros_esp32_extensions/sdkconfig.defaults:3]: sorry, I meant a link to somewhere where is explained which was this issue with `CONFIG_FREERTOS_UNICORE` and why /how they have solved it.

Comment (donRaphaco) [reply microros_esp32_extensions/sdkconfig.defaults:3]: Well to be honest I never really understood the issue with `CONFIG_FREERTOS_UNICORE`.
All I know is that in the first PR #12 I made, without this setting there was a runtime error even before the `app_main` function was executed.
I have no idea what the actual problem was then and also don't know when and how it was fixed.
So sorry I don't have a reference for this.

Comment (pablogs9) [reply microros_esp32_extensions/sdkconfig.defaults:3]: Ok, so I will try both ESP32 + micro-ROS build tools and I will remove it from config defaults
